### PR TITLE
Add get_service_node_status rpc endpoint

### PR DESCRIPTION
**Pull request by darcys22**
Creates a rpc endpoint to query the status of the service node currently being run by lokid

addresses #233

### CODE REVIEW

**Comment by darcys22** [comment]
I haven't fully tested this yet. Could I get some feedback to see if this is in line with the teams expectations?

**Comment by darcys22** [comment]
So looks like it works, right now all that is returned is the same struct that `get_service_nodes` returns but it only is for the single service node that the lokid is running from `get_service_node_key`

seandarcy_—_sean____dev____119_77_85_139_—_ssh_sean_192_168_1_98_—_Solarized_Dark_xterm-256color_—_121×34

File: src/rpc/core_rpc_server_commands_defs.h [hunk 1]
```diff
@@ -2875,6 +2875,41 @@ constexpr char const CORE_RPC_STATUS_TX_LONG_POLL_MAX_CONNECTIONS[] = "Daemon ma
...
+      BEGIN_KV_SERIALIZE_MAP()
+        KV_SERIALIZE(include_json);
+      END_KV_SERIALIZE_MAP()
+    };
+    typedef epee::misc_utils::struct_init<request_t> request;
+
+    struct response_t
+    {
+
+      struct cryptonote::COMMAND_RPC_GET_SERVICE_NODES::response_t::entry service_node_state; // Service node registration information
```
Comment (jagerman) [inline src/rpc/core_rpc_server_commands_defs.h]: You don't need `struct` here (it's a C thing in this context that isn't needed at all in C++).

**Comment by darcys22** [comment]
updated, also added the include_json option in the request

**Pull request merged**